modules/misc/xdg-mime.nix

```
@@ -0,0 +1,45 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.xdg.mime;
+
+in
+
+{
+  options = {
+    xdg.mime.enable = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Whether to install files to support the
+        <link xlink:href="https://specifications.freedesktop.org/shared-mime-info-spec/shared-mime-info-spec-latest.html">XDG Shared MIME-info specification</link> and the
+        <link xlink:href="https://specifications.freedesktop.org/mime-apps-spec/mime-apps-spec-latest.html">XDG MIME Applications specification</link>.
+      '';
+    };
+  };
+
+  config = mkIf config.xdg.mime.enable {
+    home.packages = [
+      # this package also installs some useful data, as well as its utilities
+      pkgs.shared-mime-info
+    ];
+
+    home.sessionVariables = {
+      XDG_DATA_DIRS = "$XDG_DATA_DIRS:${config.home.profileDirectory}/share";
+    };
+
+    home.extraProfileCommands = ''
+      if [ -w $out/share/mime ] && [ -d $out/share/mime/packages ]; then
+          XDG_DATA_DIRS=$out/share PKGSYSTEM_ENABLE_FSYNC=0 ${pkgs.buildPackages.shared-mime-info}/bin/update-mime-database -V $out/share/mime > /dev/null
```

Can split over multiple lines for readability, for example
```suggestion
          XDG_DATA_DIRS=$out/share \
          PKGSYSTEM_ENABLE_FSYNC=0 \
          ${pkgs.buildPackages.shared-mime-info}/bin/update-mime-database \
            -V $out/share/mime > /dev/null
```